test/postprocess.test.js

```diff
@@ -115,23 +117,28 @@ describe("postprocessing/image.js", () => {
         // validate errors
         assert.ok(result.renditionErrors === undefined);
 
-        assert.equal(events.length, 1);
-        assert.equal(events[0].type, "rendition_created");
-        assert.equal(events[0].rendition.fmt, "jpg");
-        assert.equal(events[0].metadata["tiff:imageWidth"], 10);
-        assert.equal(events[0].metadata["tiff:imageHeight"], 6);
-        assert.equal(events[0].metadata["dc:format"], "image/jpeg");
+        assert.strictEqual(events.length, 1);
+        assert.strictEqual(events[0].type, "rendition_created");
+        assert.strictEqual(events[0].rendition.fmt, "jpg");
+        assert.strictEqual(events[0].metadata["tiff:imageWidth"], 10);
+        assert.strictEqual(events[0].metadata["tiff:imageHeight"], 6);
+        assert.strictEqual(events[0].metadata["dc:format"], "image/jpeg");
 
         assert.ok(RENDITION_JPG.equals(uploadedRenditions["/MyRendition.jpeg"]));
 
         // check metrics
+
+
         await MetricsTestHelper.metricsDone();
-        assert.equal(receivedMetrics[0].eventType, "rendition");
-        assert.equal(receivedMetrics[0].callbackProcessingDuration + receivedMetrics[0].postProcessingDuration, receivedMetrics[0].processingDuration);
-        assert.equal(receivedMetrics[1].eventType, "activation");
-        assert.equal(receivedMetrics[1].callbackProcessingDuration + receivedMetrics[1].postProcessingDuration, receivedMetrics[1].processingDuration);
-        assert.equal(receivedMetrics[0].imagePostProcess, true);
-        assert.equal(receivedMetrics[1].imagePostProcess, true);
+
+        assert.strictEqual(receivedMetrics[0].eventType, "rendition");
+        assert.strictEqual(receivedMetrics[0].callbackProcessingDuration + receivedMetrics[0].postProcessingDuration, receivedMetrics[0].processingDuration);
+        assert.strictEqual(receivedMetrics[1].eventType, "activation");
+        assert.strictEqual(receivedMetrics[1].callbackProcessingDuration + receivedMetrics[1].postProcessingDuration, receivedMetrics[1].processingDuration);
+
+        // queued metrics may "flatten metrics" and turn booleans true/false into 1/0
```

Comment:
interesting find. I think we should always flatten the metrics regardless of if they are sent in the queue or not? wdyt?

looks like they may always flatten actually since they are always sent through the queue class: https://github.com/adobe/node-openwhisk-newrelic/blob/master/lib/newrelic.js#L151

Reply:
Yeah. I think mostly the tests didn't pick up the change. But to be sure, I updated the tests to accept both.